mail_template_multi_company/models/mail_template.py

```diff
@@ -0,0 +1,18 @@
+# Copyright 2017 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class MailTemplate(models.Model):
+
+    _inherit = 'mail.template'
+
+    company_id = fields.Many2one(
+        'res.company',
+        'Company',
+        default=lambda self: self.env['res.company']._company_default_get(
+            'mail.template'
+        ),
```

Comment:
We should not set the company by default on all existing mail templates when installing this module.
Because many standard template need not be specialized by company.

OTOH, when this module is installed it makes sense that *new* templates being created by users have a company set by default.